scripts/sign.py

```diff
@@ -63,16 +123,55 @@ def main():
     h.update(shdr_uuid)
     h.update(shdr_version)
     h.update(img)
-    sig = signer.sign(h)
-
-    f = open(args.out, 'wb')
-    f.write(shdr)
-    f.write(h.digest())
-    f.write(sig)
-    f.write(shdr_uuid)
-    f.write(shdr_version)
-    f.write(img)
-    f.close()
+    img_digest = h.digest()
+
+    def write_image_with_signature(sig):
+        with open(args.outf, 'wb') as f:
+            f.write(shdr)
+            f.write(img_digest)
+            f.write(sig)
+            f.write(shdr_uuid)
+            f.write(shdr_version)
+            f.write(img)
+
+    def sign_ta():
+        if not key.has_private():
+            logger.error('Provided key cannot be used for signing, ' +
+                         'please use offline-signing mode.')
+        else:
+            signer = PKCS1_v1_5.new(key)
+            sig = signer.sign(h)
+            write_image_with_signature(sig)
+            logger.info('Successfully signed application.')
+
+    def generate_digest():
+        with open(args.digf, 'wb+') as digfile:
+            digfile.write(base64.b64encode(img_digest))
+
+    def stitch_ta():
+        try:
+            with open(args.sigf, 'r') as sigfile:
+                sig = base64.b64decode(sigfile.read())
+        except IOError:
+            logger.error('No signature file found. Please sign\n %s\n' +
+                         'offline and put the signature next to it as \n %s',
```

Comment:
"(...) and put the signature next to it (...)" sounds strange.

Reply:
I clarified the message.